### PR TITLE
ft: refactor ingest processing pipeline

### PR DESCRIPTION
**Pull request by hauleth**
In addition to the fixed pipeline this commit adds Telemetry metrics to ingestion pipeline. For now it is rather crude approach where there is single metric for everything happening (no detailed view), but that will be added in near future.

### CODE REVIEW

File: lib/telemetry.ex [hunk 1]
```diff
@@ -69,7 +69,21 @@ defmodule Logflare.Telemetry do
       summary("broadway.processor.stop", unit: {:native, :millisecond})
     ]
 
-    phoenix_metrics ++ database_metrics ++ vm_metrics ++ cache_metrics ++ broadway_metrics
+    application_metrics = [
+      summary("logflare.logs.process.stop.duration",
```
Comment (Ziinc) [inline lib/telemetry.ex:73]:
```suggestion
      summary("logflare.logs.processor.ingest.stop.duration",
```

File: lib/telemetry.ex [hunk 2]
```diff
@@ -69,7 +69,21 @@ defmodule Logflare.Telemetry do
       summary("broadway.processor.stop", unit: {:native, :millisecond})
     ]
 
-    phoenix_metrics ++ database_metrics ++ vm_metrics ++ cache_metrics ++ broadway_metrics
+    application_metrics = [
```
Comment (Ziinc) [inline lib/telemetry.ex:72]: maybe additionally, `counter("logflare.logs.processor.ingest.exception.count")`

File: lib/logflare/logs/processor.ex [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+  @spec ingest([map()], module(), Logflare.Source.t()) :: :ok | {:error, term()}
+  def ingest(data, processor, source) when is_list(data) do
+    metadata = %{
+      processor: processor,
+      source: source
+    }
+
+    :telemetry.span([:logflare, :logs, :process], metadata, fn ->
+      batch = processor.handle_batch(data, source)
```
[the remark this answers is not in the document]
Comment (hauleth) [reply lib/logflare/logs/processor.ex:15]: There should be no dynamic part in the Telemetry span name, all that should go to the metadata. Splitting span into 2 IMHO makes no sense, if anything I would do nested span.

File: lib/logflare_web/controllers/log_controller.ex [hunk 4]
```diff
@@ -61,136 +60,134 @@ defmodule LogflareWeb.LogController do
...
-    log_params = Map.drop(log_params, ["source", "timestamp", "id"])
-    ingest_and_render(conn, [log_params], source)
+    log_params
+    |> Map.drop(["source", "timestamp", "id"])
+    |> List.wrap()
+    |> Processor.ingest(Logs.Raw, source)
+    |> handle(conn)
   end
 
   def syslog(%{assigns: %{source: source}} = conn, %{"batch" => batch}) when is_list(batch) do
```
[the remark this answers is not in the document]
Comment (hauleth) [reply lib/logflare_web/controllers/log_controller.ex:97]: We probably could refactor it even further, but so far I went with minimal changes that keep the nature of the original code.

File: lib/logflare/logs/processor.ex (hunk 3 above)
[the remark this answers is not in the document]
Comment (Ziinc) [reply lib/logflare/logs/processor.ex:30]: Understand, yes nesting would be a possibility. Let's see

**Review by Ziinc — APPROVED** [review]

**Pull request merged**